datastore.go

```diff
@@ -104,3 +107,11 @@ func GetBackedHas(ds Datastore, key Key) (bool, error) {
 		return false, err
 	}
 }
+
+type Transaction interface {
```

Comment:
let's call this a `Batch` because `Transaction` implies transactional semantics, with atomicity and potentially rollback guarantees. None of that here. We could add a:

```go
type Batch interface {
    IsTransactional() bool
}
```
to check whether a batch is transactional or not.